lib/phoenix/socket/transport.ex

```diff
@@ -388,6 +388,33 @@ defmodule Phoenix.Socket.Transport do
     end
   end
 
+  def extract(conn, opts) do
```

Comment:
There were strong feelings against providing access to the conn directly to prevent things like cookies and other private data to be lifted off during this phase. After a lot of conversation it was determined that we can solve mostly all problems with access to peer data and x-headers.

Reply:
@mobileoverlord, fair enough. I recall some resistance against access to things like cookies too.

One use case (if I am not mistaken) is not solved by just handing over `x-*` headers and peer information. For multitenant scenario's, I could imagine one will find the host information (schema, host, port basically) useful too.
(At least I think I would in order to lookup host based information without something in front like nginx that sets x-forwarded-host, port and proto; even if you do use a reverse proxy in production, I think in development it is quite common to not have one in front)

Reply:
For the above, I think we could also support the `:uri` option to extract the above into into a %URI{}